docs(changelog): add example as approved in #1604

gacallea opened the pull request:
matching https://github.com/gacallea/freesound-credits/blob/ac12bc0cf2df64df935f8f1fbd6e2a3888f22ca6/CHANGELOG.md

I would add some wording and screenshot but I am keeping simnple for now, matching the docs style.

cheers!

MarcoIeni commented:
The config page is already huge so I don't want to add changelog configuration examples there. Imo we should create a new page similar to [this](https://git-cliff.org/docs/templating/examples/) one. Do you want to work on it?

gacallea commented:
I agree with you. I didn’t know if I could.
I am on it!
  
On Thu, Aug 15, 2024 at 8:15 AM, Marco Ieni ***@***.***&gt; wrote:  The config page is already huge so I don't want to add changelog configuration examples there. Imo we should create a new page similar to this one. Do you want to work on it?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

gacallea commented:
@MarcoIeni ignore pushes for now. I will let you know when it's ready.

gacallea commented:
@MarcoIeni I believe I am done phew....

In the end I preferred splitting the single file into its own category and adding sections.

Examples wise: this was the hard part to fix/adjust/test rinse and repeat. I decided to include the three most interesting ones instead of copying over from git-cliff. Some of them may require things you don't have in your schema and if people really want what I considered redundant and uninteresting, they can be my guest... it was a PITA :P

I also added some language conf to docusaurus because highlight of toml and json wasn't working in code blocks.

Let me know what you think!

gacallea commented:
ps: I have not being able to avoid the messed menu in Examples (see on the right, it includes mdx-code-block when it shouldn't ) but I believe an up to date docusaurus will fix that.

gacallea commented:
hi @MarcoIeni , I see some things fail. Link I cabn't do much about it since the examples folder doesn't exist yet. Anything else let me know if you need anything from me.

MarcoIeni commented on website/docusaurus.config.js:
```diff
@@ -157,6 +157,7 @@ const config = {
       prism: {
         theme: lightCodeTheme,
         darkTheme: darkCodeTheme,
+       additionalLanguages: ['toml', 'json'],
```
```suggestion
        additionalLanguages: ['toml', 'json'],
```

gacallea replied:
sorry about that!

MarcoIeni commented:
> I believe an up to date docusaurus will fix that.

I updated the branch so that we are using a more recent version 👍
Running `npm i` brings you to 3.5.1

I'm having a quick look now, but I don't have time to do a full review. Maybe later today 👍

MarcoIeni commented:
Thanks for adding json and toml syntax highlight btw. It looks awesome 🤩

gacallea commented:
trying to fix the broken toc on examples. requesting backup https://github.com/facebook/docusaurus/discussions/10409

gacallea commented:
> trying to fix the broken toc on examples. requesting backup [facebook/docusaurus#10409](https://github.com/facebook/docusaurus/discussions/10409)

not much I can do without breaking at least something. I am leaving this as is.

gacallea commented:
@MarcoIeni I would like to add two/three more examples:

1 - default (showing release-plz default config/output)
2 - [github](https://github.com/orhun/git-cliff/blob/main/examples/github.toml) and/or [github+keepachangelog](https://github.com/orhun/git-cliff/blob/main/examples/github-keepachangelog.toml)

for 1: where can I find this to easily add an example file?
for 2: this depends if release-plz uses the same *.github.* variables but I haven't been able to find them in the schema.

cheers

MarcoIeni commented:
> for 2: this depends if release-plz uses the same .github. variables but I haven't been able to find them in the schema.

Yes, I haven't implemented those yet. https://github.com/MarcoIeni/release-plz/issues/989
I plan to work on this soonish 👍

gacallea commented:
> > for 2: this depends if release-plz uses the same .github. variables but I haven't been able to find them in the schema.
> 
> Yes, I haven't implemented those yet. #989 I plan to work on this soonish 👍

ok then :) I think I am done for real now :)

let me know if you need anything else!

MarcoIeni commented:
I noticed the default changelog isn't producing the correct format. I removed the extra new line in the body but I still need to understand why the header doesn't have the new line. Maybe it's a bug 👀
<img width="523" alt="image" src="https://github.com/user-attachments/assets/04eaacaa-0d1a-4fa1-b6c4-513077553dfb">

Also, I think it's better to inline the examples in the docs instead of having them in a separate examples dir.

gacallea commented:
> I noticed the default changelog isn't producing the correct format. I removed the extra new line in the body but I still need to understand why the header doesn't have the new line. Maybe it's a bug 👀 <img alt="image" width="523" src="https://private-user-images.githubusercontent.com/11428655/358899174-04eaacaa-0d1a-4fa1-b6c4-513077553dfb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM5NzgzODcsIm5iZiI6MTcyMzk3ODA4NywicGF0aCI6Ii8xMTQyODY1NS8zNTg4OTkxNzQtMDRlYWFjYWEtMGQxYS00ZmExLWI2YzQtNTEzMDc3NTUzZGZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE4VDEwNDgwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzNDNhOWMxZjQ0M2I5ODFmOGJiMGE0NjIxYzBlZWI5Mzc1MWI0MWQzMjQ1NzhmNTNiNTUzMmZjMTUyY2NlODImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6W9Tww5uY58UCW4sqjjpvsl0GkQpTmfj7jEErscBAO4">
> 
> Also, I think it's better to inline the examples in the docs instead of having them in a separate examples dir. In this way, people do less back-and-forth

Fair enough :)

I added the extra line because of that "bug" and because markdown wants lines in between headers and such. I do so in my config as well.

Regarding the examples dir vs inline, I also tend to agree if it is clean enough. I just did what git-cliff was doing.

I am going to check these changes later, if there is anything I need to do please let me know.

gacallea commented:
ps: I have an idea for inlining the conf:

just adding a toml code block in the same fashion as the output ones (folded):

- Tomo configuration
- Raw output
- Rendered output

gacallea commented:
> ps: I have an idea for inlining the conf:
> 
> just adding a toml code block in the same fashion as the output ones (folded):
> 
> * TOML configuration
> * Raw output
> * Rendered output

I implemented this, let me know if it's good now :)

gacallea commented:
@MarcoIeni regarding markdown-cli failing: I tried fixing those header coming from `mdx-code-block` code but there is no way around it. (see above discussion on docusaurus repo). I'd opt to remove those entirely. What do you reckon?

MarcoIeni commented on website/docs/configuration/changelog_examples.md:
```diff
@@ -0,0 +1,643 @@
+{% endfor %}\n
+"""
+
+footer = """
```
release-plz doesn't support the footer. That's because git-cliff only recently added the ability to have templates there so at the beginning I didn't add support for it.
I created https://github.com/MarcoIeni/release-plz/issues/1621

Also, I'm not a fan of commit_parsers. With those commit parsers a commit like `fix: add abc` will be identified as "Changed".

Since (except for the footer part), this Keep A Changelog config is very similar to the default one, imo we can delete this example entirely. What do you think?

gacallea replied:
footer: sorry I didn't know and didn't check. my bad. I don't even like having the footer.

keep a changelog can go then :)

commit_parsers: I am new to this and I liked the bling bling, it's your call tho :)

let me know if I should change it or if you arew taking care of it.

gacallea replied:
done, let me know if you prefer to keep mdx-blocks . they were mostly an annoyance.

gacallea replied:
ps: feel free to change the commit_parsers too, but I think it's good to give alternatives examples otherwise why would even have this section?

gacallea replied:
alternatively: redefining the parsers grouping to your liking?

MarcoIeni replied:
For the mdx-blocks we can disable the lint 👍
I'll think about it

gacallea replied:
let me know, I have currently removed them.

MarcoIeni commented on website/docs/configuration/tips-and-tricks.md:
````diff
@@ -0,0 +1,89 @@
+### {{ group | striptags | trim | upper_first }}
+```
+
+### Use GitHub PR labels as groups
````
we don't have this feature yet.

MarcoIeni commented on website/docs/configuration/changelog_examples.md:
```diff
@@ -0,0 +1,440 @@
+    ## [unreleased]
+{% endif %}\
+{% if previous %}\
+    {% if previous.commit_id and commit_id %}
```
https://github.com/gacallea/release-plz-contrib/blob/f05ecfe613564ab7dbdd412d71c47b236d84a4db/crates/release_plz_core/src/changelog.rs#L290
This is never populated

gacallea replied:
sorry. (see next comment)

MarcoIeni commented on website/docs/configuration/changelog_examples.md:
```diff
@@ -0,0 +1,440 @@
+body = """
+{% if version %}\
+    ## [{{ version | trim_start_matches(pat="v") }}] - {{ timestamp | date(format="%Y-%m-%d") }}
+{% else %}\
```
We never go into the `else` because release-plz always populates the `version` field. I'm removing the detailed template because it doesn't fit for release-plz. If needed, we can add it into another PR 👍

gacallea replied:
sorry about that!

truth be told I have been struggliong a lot with testing out stuff because I had to rely on `git cliff -o CHANGELOG.md` during my contribution. I am yet to find a way (even for mty personal use) to generate a changelog from scratch entirely with release-plz. Is there a command to do that?

MarcoIeni replied:
>  to generate a changelog from scratch entirely with release-plz. Is there a command to do that?

no 😂 https://github.com/MarcoIeni/release-plz/issues/1009

> sorry about that!



Don't worry :)

MarcoIeni commented:
I reverted back the config.md file. I'm not sure it's the right thing to do because changelog is a part of the configuration but at least in this way we don't break existing links (maybe there's a way to redirect, not sure). Anyway for now it should be good enough 👍

gacallea commented:
> I reverted back the config.md file. I'm not sure it's the right thing to do because changelog is a part of the configuration but at least in this way we don't break existing links (maybe there's a way to redirect, not sure). Anyway for now it should be good enough 👍

it is your project/site and the choice is ultimately yours. i did my best categorizing things and fixed all links `npm run build` complained about. unless you are worried about SEO here.

MarcoIeni reviewed (approved):
Yay, we did it! Thanks 🎉

gacallea commented:
yay! thank you Marco for this amazing tool! I did my best but you had to comb over lots! Sorry it is my first contrib with Rust (even if just docs). I'll do better next time :P

(not right now but I would like to have a go at #1009 when I finish The Book -- ssssh don't tell anyone!)

MarcoIeni commented:
It was a great contribution. I'm happy you did your first Rust contribution in release-plz 🥳

That issue is really hard! If you want a less hardcore experience check [good-first-issues](https://github.com/MarcoIeni/release-plz/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) 👍
